dashboard/src/components/AppView/AccessURLItem.tsx

```diff
@@ -0,0 +1,84 @@
+import * as React from "react";
+
+import { IResource, IServiceSpec, IServiceStatus } from "../../shared/types";
+import "./AccessURLItem.css";
+
+interface IAccessURLItem {
+  service: IResource;
+}
+
+interface IAccessURLState {
+  isLink: boolean;
+  URLs: string[];
+}
+
+class AccessURLItem extends React.Component<IAccessURLItem, IAccessURLState> {
+  public state: IAccessURLState = {
+    isLink: false,
+    URLs: [],
+  };
+
+  public componentDidMount() {
```

Comment:
See my comment about not needing state and related to that, I do not think that you need to use componentDidMount, you can just have two methods, or two getters, or one returning a struct.

Reply:
Agreed, this can just be evaluated in render